### PR TITLE
feat(resize-observer-browser): updating type definition

### PR DESCRIPTION
**Pull request by willmendesneto**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [x] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/resize-observer-browser/index.d.ts
Comment (willmendesneto) [inline types/resize-observer-browser/index.d.ts]: More details of it in https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver/ResizeObserver

File: types/resize-observer-browser/index.d.ts
Comment (chivesrs) [inline types/resize-observer-browser/index.d.ts]: Rename the type of options to `ResizeObserverOptions`, as this follows the spec at https://drafts.csswg.org/resize-observer/#dictdef-resizeobserveroptions.

File: types/resize-observer-browser/index.d.ts
Comment (chivesrs) [inline types/resize-observer-browser/index.d.ts]: Can you use single quotes in the `@default` section?
`'content-box'`

File: types/resize-observer-browser/index.d.ts
Comment (chivesrs) [inline types/resize-observer-browser/index.d.ts]: Rename to `ResizeObserverSize` to match spec https://drafts.csswg.org/resize-observer-1/#resizeobserversize.

File: types/resize-observer-browser/index.d.ts
Comment (chivesrs) [inline types/resize-observer-browser/index.d.ts]: Re order these two so `inlineSize` is first (matches the spec).

**Review by chivesrs — CHANGES_REQUESTED** [review]
Some minor changes requested.

File: types/resize-observer-browser/index.d.ts
Comment (wffurr) [reply types/resize-observer-browser/index.d.ts]: Also, why the different style of specifying the function type for the observe member vs. using the member shorthand?

Comment (chivesrs) [reply types/resize-observer-browser/index.d.ts]: Good catch. This shouldn't be using the syntax you changed to.

should read:

`observe(target: Type, options?: Type): void;`

File: types/resize-observer-browser/index.d.ts
Comment (willmendesneto) [reply types/resize-observer-browser/index.d.ts]: sure thing. applying that now

File: types/resize-observer-browser/index.d.ts
Comment (willmendesneto) [reply types/resize-observer-browser/index.d.ts]: Good catch! Applying these changes now

File: types/resize-observer-browser/index.d.ts
Comment (willmendesneto) [reply types/resize-observer-browser/index.d.ts]: 👍

File: types/resize-observer-browser/index.d.ts
Comment (willmendesneto) [reply types/resize-observer-browser/index.d.ts]: done

**Comment by willmendesneto** [comment]
@chivesrs @wffurr  I already applied the given feedback, thanks for that!

Could you please review again, please?

**Review by chivesrs — APPROVED** [review]

**Comment by willmendesneto** [comment]
Hey folks. Any idea when we can get these changes merged and published? Just looking forward to use this type definition directly in my projects 😊

**Comment by chivesrs** [comment]
I can't merge it. It has to be someone that has merge rights on the DefinitelyTyped repository. My experience is anywhere from 1 day to 3 weeks, depending on how busy the TypeScript team at Microsoft is.

**Comment by willmendesneto** [comment]
@rbuckton could you help me on that, please?

**Pull request merged**
